sdk/azcore/runtime/policy_bearer_token.go

```diff
@@ -97,7 +97,9 @@ func (b *BearerTokenPolicy) authenticateAndAuthorize(req *policy.Request) func(p
 		as := acquiringResourceState{p: b, req: req, tro: tro}
 		tk, err := b.mainResource.Get(as)
 		if err != nil {
-			return err
+			// consider this error non-retriable because if it could be resolved by
+			// retrying authentication, the credential would have done so already
+			return errorinfo.NonRetriableError(err)
```

Comment:
#25021 makes this an almost no-op for the common case of errors returned by `azidentity`

Reply:
Someone has to throw away that pointer 😁